pkg/tfgen/docs.go

```diff
@@ -502,13 +506,20 @@ const (
 	sectionImports             = 5
 )
 
-func (p *tfMarkdownParser) parseSupplementaryExamples() (string, error) {
+func (p *tfMarkdownParser) readFile(name string) ([]byte, error) {
+	if p.readFileFunc != nil {
```

Comment:
Can we add a comment that makes it clear this if branch is hit by test cases only?